code/client/client.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+#include <yarp/os/all.h>
+
+using namespace std;
+using namespace yarp::os;
+
+int main(int argc, char * argv[])
+{
+	if(argc<2){
+		cout << "Please supply a port name for client" << endl;
+		return 1;
+	}
+
+	const char *charName = argv[1];
+
+    Network yarp;	//initialize yarp
+    Semaphore sem;
+    Port portP;
+    portP.open(charName);
+    
```

Comment:
@kawasetsu `RpcServer` and `RpcClient` are mere specialization of `Port`, hence using `Port` is perfectly doable, even though making it clear that you're gonna use it in a given context would be better.